src/figuresClasses.ts

```diff
@@ -1,19 +1,76 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0
+      || this.a >= (this.b + this.c)
+      || this.b >= (this.a + this.c)
+      || this.c >= (this.b + this.a)) {
+      throw new Error('check the sides of the triangle!');
+    }
```

Comment:
You check that input values are bigger than 0 for each class
Will be better to create separate function, that will accept initial parameters, check it and return the corresponding error if not
And you can call this one general function in each class